tools/qa/buildkite_static_code_analysis2.sh

```diff
@@ -1,14 +1,37 @@
 #!/usr/bin/env bash
 
+dir="tools/qa"
+
 if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then
-    source tools/qa/buildkite_common.sh
+    source $dir/buildkite_common.sh
     get_ancestor  # Writes $ANCESTOR_SHA variable.
 
-    echo "--- Running trapdoors tests"
+    echo "--- Prep working directory"
+    rm -rf *_pr.tar.gz *_ancestor.tar.gz
+    ./cleanfiles.sh
+
+    echo "--- Unpack PR build from previous step"
+    buildkite-agent artifact download horton_pr.tar.gz .
+    tar xvf horton_pr.tar.gz
+
+    echo "--- Build refatoms"
+    rm -rf data/refatoms/*.h5 data/refatoms/*.tar.bz2
+    make -C data/refatoms/
+
+    echo "--- Running trapdoors tests on PR branch"
+    $dir/simulate_trapdoor_pr.py -vA $ANCESTOR_SHA -o feature $dir/trapdoor_pylint.py
+
+    echo "--- Unpack ancestor build from previous step"
```

Comment:
Minor: it would be safe to unpack the build after checking out the source code. (similar situation in other scripts)